src/manifests_workflow/input_manifests.py

```diff
@@ -11,6 +11,8 @@
 from abc import abstractmethod
 from typing import Dict, List, Type, Union
 
+import ruamel.yaml
```

Comment:
What is the reason to use ruamel.yaml without using the default yaml module?

Reply:
Hey @peterzhuamazon there is no default YAML module, the comparison I did was between [PyYAML](https://pyyaml.org/wiki/PyYAMLDocumentation) and [ruamel.yaml](https://pypi.org/project/ruamel.yaml/) and I see `ruamel.yaml` is working better when dealing with yaml parsing.
1) Using `ruamel.yaml` it's easy to override the methods like `explicit_start`, `preserve_quotes` and right `indent` which honors `yamllint` and gh  workflow `.yml` rules.
2) `ruamel.yaml` also does not change the [on](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/increment-plugin-versions.yml#L4) key to boolean (default the library considers `on` as boolean) which is not easy with PyYAML, need some extra logic to create constructer and override methods.
https://stackoverflow.com/questions/36463531/pyyaml-automatically-converting-certain-keys-to-boolean-values
3) [---](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/increment-plugin-versions.yml#L1), this is also preserved with `ruamel.yaml` easily compared to `PyYAML`, without this the GH and yamllint flags it as error.

Reply:
Since we are running our python code on win/macos now can you also confirm these two can be installed correctly on win/macos? We have cases before where some modules are not compatible on those platforms.

Thanks.

Reply:
And yes I mean the PyYAML module as it is widely used.

Reply:
Got it, I see windows python tests pass, but I will confirm if this package has any restrictions to windows.
Thanks @peterzhuamazon

Reply:
Mac it works with as all my tests and development are from mac (with jdk14)
from website i see its as [os independent](https://pypi.org/project/ruamel.yaml.cmd/)
Some more useful details: https://snyk.io/advisor/python/ruamel-yaml

Reply:
@prudhvigodithi Do you think the default PyYaml should be replaced by this library?
If yes can you create an issue since you already have good points noted above?
Thanks!

Reply:
Yes I can @gaiksaya, some helpful info
cookiecutter migration to `ruamel.yaml` https://github.com/cookiecutter/cookiecutter/issues/557
`ruamel.yaml` difference with `PyYAML`: https://yaml.readthedocs.io/en/latest/pyyaml.html#